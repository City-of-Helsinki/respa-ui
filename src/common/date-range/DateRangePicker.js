import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import DayPicker, { DateUtils } from 'react-day-picker';
import MomentLocaleUtils from 'react-day-picker/moment';
import 'react-day-picker/lib/style.css';

import injectT from '../../../app/i18n/injectT';
import {
  findMatchingPeriod,
  validateRange,
  DatePickerValidationError,
  isRange,
  trimAvailabilityPeriods,
} from './utils';
import { createNotification } from '../notification/utils';
import { NOTIFICATION_TYPE } from '../notification/constants';

const DateRangePicker = ({
  fullDay,
  startingDays,
  openingPeriods,
  reservations,
  onRangeSelect,
  renderChildComponent,
  initialMonth = new Date(),
  onMonthChange,
  t,
  locale,
}) => {
  const [selectedDateRange, setSelectedDateRange] = useState({
    from: undefined,
    to: undefined,
  });

  useEffect(() => {
    setSelectedDateRange({
      from: undefined,
      to: undefined,
    });
  }, [fullDay]);

  const handleDayClick = (day, modifiers = {}) => {
    // Day not available: don't permit selection.
    if (!modifiers.available) {
      return;
    }

    // Not a starting day and no starting day selected yet: make no selection.
    if (!modifiers.startingDay && !selectedDateRange.from) {
      return;
    }

    // Starting day selected, but attempting to select day before the starting day: don't update selection
    if (
      selectedDateRange.from
      && !selectedDateRange.to
      && day < selectedDateRange.from
    ) {
      return;
    }

    // Both starting day and ending day selected: reset selection to empty.
    if (selectedDateRange.from && selectedDateRange.to) {
      setSelectedDateRange({ from: undefined, to: undefined });
      return;
    }

    const range = DateUtils.addDayToRange(day, selectedDateRange);

    const period = findMatchingPeriod(range, openingPeriods);
    period
      && range.from
      && range.from.setHours(period.startingTime.hours, period.startingTime.min);
    period
      && range.to
      && range.to.setHours(period.endingTime.hours, period.endingTime.min);
    const minDuration = period ? period.minDuration : undefined;
    const maxDuration = period ? period.maxDuration : undefined;

    setSelectedDateRange(range);

    if (isRange(range)) {
      const validationErrors = validateRange(range, {
        startingDays,
        openingPeriods,
        reservations,
        fullDay,
      });

      if (validationErrors.length === 0) {
        onRangeSelect && onRangeSelect(range);
        return;
      }

      validationErrors.forEach((error) => {
        switch (error) {
          case DatePickerValidationError.RANGE_TOO_LONG:
            createNotification(
              NOTIFICATION_TYPE.WARNING,
              t('DateRangePicker.info.maxPeriodText', {
                duration: maxDuration,
                durationUnit: t(
                  `common.unit.time.${period.durationUnit}.plural`,
                ),
              }),
            );
            break;
          case DatePickerValidationError.RANGE_TOO_SHORT:
            createNotification(
              NOTIFICATION_TYPE.WARNING,
              t('DateRangePicker.info.minPeriodText', {
                duration: minDuration,
                durationUnit: t(
                  `common.unit.time.${period.durationUnit}.plural`,
                ),
              }),
            );
            break;
          case DatePickerValidationError.OVERLAPPING_RESERVATION:
            createNotification(
              NOTIFICATION_TYPE.WARNING,
              t('DateRangePicker.info.overlappingReservation'),
            );
            break;
          case DatePickerValidationError.INVALID_STARTING_DAY:
            createNotification(
              NOTIFICATION_TYPE.WARNING,
              t('DateRangePicker.info.invalidStartingDay'),
            );
            break;
          case DatePickerValidationError.OUTSIDE_OPENING_PERIODS:
            createNotification(
              NOTIFICATION_TYPE.WARNING,
              t('DateRangePicker.info.outsideOpeningPeriods'),
            );
            break;
          default:
            break;
        }
      });
    }
  };

  const modifiers = {
    available: trimAvailabilityPeriods(openingPeriods),
    reserved: reservations,
    startingDay: startingDays,
    reservationStart: !fullDay
      ? reservations.map(reservation => reservation.from)
      : undefined,
    reservationEnd: !fullDay
      ? reservations.map(reservation => reservation.to)
      : undefined,
    selectionStart: !fullDay ? selectedDateRange.from : undefined,
    selectionEnd: !fullDay ? selectedDateRange.to : undefined,
  };

  return (
    <div className="app-DateRangePicker">
      <DayPicker
        initialMonth={initialMonth}
        locale={locale}
        localeUtils={MomentLocaleUtils}
        modifiers={modifiers}
        numberOfMonths={2}
        onDayClick={handleDayClick}
        onMonthChange={date => onMonthChange && onMonthChange(date)}
        selectedDays={selectedDateRange}
        showWeekNumbers
      />
      {renderChildComponent && renderChildComponent(selectedDateRange)}
    </div>
  );
};

DateRangePicker.propTypes = {
  fullDay: PropTypes.bool,
  startingDays: PropTypes.arrayOf(PropTypes.instanceOf(Date)).isRequired,
  openingPeriods: PropTypes.arrayOf(
    PropTypes.shape({
      from: PropTypes.instanceOf(Date),
      to: PropTypes.instanceOf(Date),
      durationUnit: PropTypes.string.isRequired,
      minDuration: PropTypes.number.isRequired,
      maxDuration: PropTypes.number.isRequired,
      startingTime: PropTypes.shape({
        hours: PropTypes.number,
        min: PropTypes.number,
      }).isRequired,
      endingTime: PropTypes.shape({
        hours: PropTypes.number,
        min: PropTypes.number,
      }).isRequired,
    }),
  ).isRequired,
  reservations: PropTypes.arrayOf(
    PropTypes.shape({
      from: PropTypes.instanceOf(Date),
      to: PropTypes.instanceOf(Date),
    }),
  ).isRequired,
  onRangeSelect: PropTypes.func,
  renderChildComponent: PropTypes.func,
  initialMonth: PropTypes.instanceOf(Date),
  onMonthChange: PropTypes.func,
  t: PropTypes.func.isRequired,
  locale: PropTypes.string.isRequired,
};

export default injectT(DateRangePicker);
