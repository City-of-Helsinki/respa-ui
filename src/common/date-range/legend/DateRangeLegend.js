import React from 'react';
import PropTypes from 'prop-types';

import injectT from '../../../../app/i18n/injectT';

const DateRangeLegend = (props) => {
  const { t } = props;
  return (
    <div className="app-DateRangeLegend">
      <div className="app-DateRangeLegend__legend-item">
        <div className="app-DateRangeLegend__color-box app-DateRangeLegend__color-box--available">01</div>
        <p>{t('DateRangeLegend.available')}</p>
      </div>

      <div className="app-DateRangeLegend__legend-item">
        <div className="app-DateRangeLegend__color-box app-DateRangeLegend__color-box--start">01</div>
        <p>{t('DateRangeLegend.availableStartingDay')}</p>
      </div>

      <div className="app-DateRangeLegend__legend-item">
        <div className="app-DateRangeLegend__color-box app-DateRangeLegend__color-box--reserved">01</div>
        <p>{t('DateRangeLegend.reserved')}</p>
      </div>

      <div className="app-DateRangeLegend__legend-item">
        <div className="app-DateRangeLegend__color-box app-DateRangeLegend__color-box--selection">01</div>
        <p>{t('DateRangeLegend.selection')}</p>
      </div>

      <div className="app-DateRangeLegend__legend-item">
        <div className="app-DateRangeLegend__color-box app-DateRangeLegend__color-box--unavailable">01</div>
        <p>{t('DateRangeLegend.unavailable')}</p>
      </div>
    </div>
  );
};

DateRangeLegend.propTypes = {
  t: PropTypes.func.isRequired,
};

export default injectT(DateRangeLegend);
