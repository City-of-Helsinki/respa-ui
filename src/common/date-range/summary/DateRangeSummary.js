import React from 'react';
import PropTypes from 'prop-types';

import iconCalendar from '../../../../app/assets/icons/calendar.svg';
import iconClock from '../../../../app/assets/icons/clock-o.svg';
import {
  formatDateRange,
  formatTimeRange,
  getDateRangeDuration,
} from '../utils';
import injectT from '../../../../app/i18n/injectT';

const DateRangeSummary = ({
  startDate, endDate, fullDay, t,
}) => {
  return (
    <div className="app-DateRangeSummary">
      <b>{t('DateRangeSummary.title')}</b>
      <div className="app-DateRangeSummary__summaryDetails">
        <div>
          <img
            alt="dateRangeSummaryCalendarIcon"
            className="app-DateRangeSummary__icon"
            src={iconCalendar}
          />
          <span>{formatDateRange(startDate, endDate)}</span>
        </div>
        <div>
          <img
            alt="dateRangeSummaryClockIcon"
            className="app-DateRangeSummary__icon"
            src={iconClock}
          />
          <span>{formatTimeRange(startDate, endDate)}</span>
        </div>
        <div>
          {/* TODO: get the correct svg for this icon */}
          <img
            alt="dateRangeSummaryClockIcon"
            className="app-DateRangeSummary__icon"
            src={iconClock}
          />
          <span>
            {startDate && endDate
              ? t('DateRangeSummary.days', {
                days: getDateRangeDuration(startDate, endDate, fullDay),
              })
              : '-'}
          </span>
        </div>
      </div>
    </div>
  );
};

DateRangeSummary.propTypes = {
  startDate: PropTypes.instanceOf(Date),
  endDate: PropTypes.instanceOf(Date),
  fullDay: PropTypes.bool,
  t: PropTypes.func.isRequired,
};

export default injectT(DateRangeSummary);
