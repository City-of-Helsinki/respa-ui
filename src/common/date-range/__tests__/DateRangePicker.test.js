import React from 'react';
import toJson from 'enzyme-to-json';

import {
  mountWithIntl,
  shallowWithIntl,
} from '../../../../app/utils/testUtils';
import DateRangePicker from '../DateRangePicker';
import DateRangeSummary from '../summary/DateRangeSummary';
import DateRangeLegend from '../legend/DateRangeLegend';

const mockProps = {
  fullDay: false,
  maxDuration: 7,
  minDuration: 2,
  openingPeriods: [
    {
      from: new Date(2017, 11, 5),
      to: new Date(2100, 1),
      maxDuration: 7,
      minDuration: 2,
      startingTime: { hours: 15, min: 0 },
      endingTime: { hours: 11, min: 0 },
      durationUnit: 'hour',
    },
  ],
  reservations: [
    {
      from: new Date(2017, 11, 26),
      to: new Date(2017, 11, 28),
    },
  ],
  startingDays: [
    new Date(2017, 11, 10),
    new Date(2017, 11, 15),
    new Date(2017, 11, 20),
  ],
  initialMonth: new Date(2017, 11, 10),
};

describe('DateRangePicker', () => {
  const getShallowWrapper = props => shallowWithIntl(<DateRangePicker {...props} />);
  const getMountedWrapper = props => mountWithIntl(<DateRangePicker {...props} />);

  test('renders normally', () => {
    const wrapper = getShallowWrapper(mockProps);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders child component', () => {
    const wrapper = getShallowWrapper({
      ...mockProps,
      renderChildComponent: selectedDateRange => (
        <>
          <DateRangeSummary
            endDate={selectedDateRange.to}
            startDate={selectedDateRange.from}
          />
          <DateRangeLegend />
        </>
      ),
    });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  const expectNoSelection = wrapper => expect(
    wrapper.find('.DayPicker').some('.DayPicker-Day--selectionStart'),
  ).toBe(false);

  const selectValidRange = (wrapper) => {
    wrapper.find('.DayPicker-Day--startingDay').at(0).simulate('click');
    wrapper.find('.DayPicker-Day--startingDay').at(1).simulate('click');
  };

  test('selection of non-starting days is not permitted', () => {
    const wrapper = getMountedWrapper(mockProps);

    wrapper.find('.DayPicker-Day').at(0).simulate('click');
    expectNoSelection(wrapper);
  });

  test('clicking after full selection is made clears selection', () => {
    const wrapper = getMountedWrapper(mockProps);

    selectValidRange(wrapper);
    wrapper.find('.DayPicker-Day--startingDay').at(2).simulate('click');
    expectNoSelection(wrapper);
  });

  test('valid value calls callback', () => {
    const mockOnRangeSelect = jest.fn();
    const wrapper = getMountedWrapper({
      ...mockProps,
      onRangeSelect: mockOnRangeSelect,
    });

    selectValidRange(wrapper);
    expect(mockOnRangeSelect).toBeCalledTimes(1);
    expect(mockOnRangeSelect).toBeCalledWith({
      from: new Date(2017, 11, 10, 15, 0),
      to: new Date(2017, 11, 15, 11, 0),
    });
  });
});
