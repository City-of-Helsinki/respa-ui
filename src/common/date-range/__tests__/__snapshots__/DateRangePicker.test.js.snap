// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker renders child component 1`] = `
<div
  className="app-DateRangePicker"
>
  <DayPicker
    canChangeMonth={true}
    captionElement={
      <Caption
        classNames={
          Object {
            "body": "DayPicker-Body",
            "caption": "DayPicker-Caption",
            "container": "DayPicker",
            "day": "DayPicker-Day",
            "disabled": "disabled",
            "footer": "DayPicker-Footer",
            "interactionDisabled": "DayPicker--interactionDisabled",
            "month": "DayPicker-Month",
            "months": "DayPicker-Months",
            "navBar": "DayPicker-NavBar",
            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
            "outside": "outside",
            "selected": "selected",
            "today": "today",
            "todayButton": "DayPicker-TodayButton",
            "week": "DayPicker-Week",
            "weekNumber": "DayPicker-WeekNumber",
            "weekday": "DayPicker-Weekday",
            "weekdays": "DayPicker-Weekdays",
            "weekdaysRow": "DayPicker-WeekdaysRow",
            "wrapper": "DayPicker-wrapper",
          }
        }
        localeUtils={
          Object {
            "formatDay": [Function],
            "formatMonthTitle": [Function],
            "formatWeekdayLong": [Function],
            "formatWeekdayShort": [Function],
            "getFirstDayOfWeek": [Function],
            "getMonths": [Function],
          }
        }
      />
    }
    classNames={
      Object {
        "body": "DayPicker-Body",
        "caption": "DayPicker-Caption",
        "container": "DayPicker",
        "day": "DayPicker-Day",
        "disabled": "disabled",
        "footer": "DayPicker-Footer",
        "interactionDisabled": "DayPicker--interactionDisabled",
        "month": "DayPicker-Month",
        "months": "DayPicker-Months",
        "navBar": "DayPicker-NavBar",
        "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
        "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
        "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
        "outside": "outside",
        "selected": "selected",
        "today": "today",
        "todayButton": "DayPicker-TodayButton",
        "week": "DayPicker-Week",
        "weekNumber": "DayPicker-WeekNumber",
        "weekday": "DayPicker-Weekday",
        "weekdays": "DayPicker-Weekdays",
        "weekdaysRow": "DayPicker-WeekdaysRow",
        "wrapper": "DayPicker-wrapper",
      }
    }
    enableOutsideDaysClick={true}
    fixedWeeks={false}
    initialMonth={2017-12-09T22:00:00.000Z}
    labels={
      Object {
        "nextMonth": "Next Month",
        "previousMonth": "Previous Month",
      }
    }
    locale="en"
    localeUtils={
      Object {
        "formatDate": [Function],
        "formatDay": [Function],
        "formatMonthTitle": [Function],
        "formatWeekdayLong": [Function],
        "formatWeekdayShort": [Function],
        "getFirstDayOfWeek": [Function],
        "getMonths": [Function],
        "parseDate": [Function],
      }
    }
    modifiers={
      Object {
        "available": Array [
          Object {
            "durationUnit": "hour",
            "endingTime": Object {
              "hours": 11,
              "min": 0,
            },
            "from": 2020-09-18T18:20:36.865Z,
            "maxDuration": 7,
            "minDuration": 2,
            "startingTime": Object {
              "hours": 15,
              "min": 0,
            },
            "to": 2100-01-31T22:00:00.000Z,
          },
        ],
        "reservationEnd": Array [
          2017-12-27T22:00:00.000Z,
        ],
        "reservationStart": Array [
          2017-12-25T22:00:00.000Z,
        ],
        "reserved": Array [
          Object {
            "from": 2017-12-25T22:00:00.000Z,
            "to": 2017-12-27T22:00:00.000Z,
          },
        ],
        "selectionEnd": undefined,
        "selectionStart": undefined,
        "startingDay": Array [
          2017-12-09T22:00:00.000Z,
          2017-12-14T22:00:00.000Z,
          2017-12-19T22:00:00.000Z,
        ],
      }
    }
    navbarElement={
      <Navbar
        classNames={
          Object {
            "body": "DayPicker-Body",
            "caption": "DayPicker-Caption",
            "container": "DayPicker",
            "day": "DayPicker-Day",
            "disabled": "disabled",
            "footer": "DayPicker-Footer",
            "interactionDisabled": "DayPicker--interactionDisabled",
            "month": "DayPicker-Month",
            "months": "DayPicker-Months",
            "navBar": "DayPicker-NavBar",
            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
            "outside": "outside",
            "selected": "selected",
            "today": "today",
            "todayButton": "DayPicker-TodayButton",
            "week": "DayPicker-Week",
            "weekNumber": "DayPicker-WeekNumber",
            "weekday": "DayPicker-Weekday",
            "weekdays": "DayPicker-Weekdays",
            "weekdaysRow": "DayPicker-WeekdaysRow",
            "wrapper": "DayPicker-wrapper",
          }
        }
        dir="ltr"
        labels={
          Object {
            "nextMonth": "Next Month",
            "previousMonth": "Previous Month",
          }
        }
        showNextButton={true}
        showPreviousButton={true}
      />
    }
    numberOfMonths={2}
    onDayClick={[Function]}
    onMonthChange={[Function]}
    pagedNavigation={false}
    renderDay={[Function]}
    renderWeek={[Function]}
    reverseMonths={false}
    selectedDays={
      Object {
        "from": undefined,
        "to": undefined,
      }
    }
    showOutsideDays={false}
    showWeekDays={true}
    showWeekNumbers={true}
    tabIndex={0}
    weekdayElement={<Weekday />}
  />
  <InjectT(DateRangeSummary) />
  <InjectT(DateRangeLegend) />
</div>
`;

exports[`DateRangePicker renders normally 1`] = `
<div
  className="app-DateRangePicker"
>
  <DayPicker
    canChangeMonth={true}
    captionElement={
      <Caption
        classNames={
          Object {
            "body": "DayPicker-Body",
            "caption": "DayPicker-Caption",
            "container": "DayPicker",
            "day": "DayPicker-Day",
            "disabled": "disabled",
            "footer": "DayPicker-Footer",
            "interactionDisabled": "DayPicker--interactionDisabled",
            "month": "DayPicker-Month",
            "months": "DayPicker-Months",
            "navBar": "DayPicker-NavBar",
            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
            "outside": "outside",
            "selected": "selected",
            "today": "today",
            "todayButton": "DayPicker-TodayButton",
            "week": "DayPicker-Week",
            "weekNumber": "DayPicker-WeekNumber",
            "weekday": "DayPicker-Weekday",
            "weekdays": "DayPicker-Weekdays",
            "weekdaysRow": "DayPicker-WeekdaysRow",
            "wrapper": "DayPicker-wrapper",
          }
        }
        localeUtils={
          Object {
            "formatDay": [Function],
            "formatMonthTitle": [Function],
            "formatWeekdayLong": [Function],
            "formatWeekdayShort": [Function],
            "getFirstDayOfWeek": [Function],
            "getMonths": [Function],
          }
        }
      />
    }
    classNames={
      Object {
        "body": "DayPicker-Body",
        "caption": "DayPicker-Caption",
        "container": "DayPicker",
        "day": "DayPicker-Day",
        "disabled": "disabled",
        "footer": "DayPicker-Footer",
        "interactionDisabled": "DayPicker--interactionDisabled",
        "month": "DayPicker-Month",
        "months": "DayPicker-Months",
        "navBar": "DayPicker-NavBar",
        "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
        "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
        "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
        "outside": "outside",
        "selected": "selected",
        "today": "today",
        "todayButton": "DayPicker-TodayButton",
        "week": "DayPicker-Week",
        "weekNumber": "DayPicker-WeekNumber",
        "weekday": "DayPicker-Weekday",
        "weekdays": "DayPicker-Weekdays",
        "weekdaysRow": "DayPicker-WeekdaysRow",
        "wrapper": "DayPicker-wrapper",
      }
    }
    enableOutsideDaysClick={true}
    fixedWeeks={false}
    initialMonth={2017-12-09T22:00:00.000Z}
    labels={
      Object {
        "nextMonth": "Next Month",
        "previousMonth": "Previous Month",
      }
    }
    locale="en"
    localeUtils={
      Object {
        "formatDate": [Function],
        "formatDay": [Function],
        "formatMonthTitle": [Function],
        "formatWeekdayLong": [Function],
        "formatWeekdayShort": [Function],
        "getFirstDayOfWeek": [Function],
        "getMonths": [Function],
        "parseDate": [Function],
      }
    }
    modifiers={
      Object {
        "available": Array [
          Object {
            "durationUnit": "hour",
            "endingTime": Object {
              "hours": 11,
              "min": 0,
            },
            "from": 2020-09-18T18:20:36.851Z,
            "maxDuration": 7,
            "minDuration": 2,
            "startingTime": Object {
              "hours": 15,
              "min": 0,
            },
            "to": 2100-01-31T22:00:00.000Z,
          },
        ],
        "reservationEnd": Array [
          2017-12-27T22:00:00.000Z,
        ],
        "reservationStart": Array [
          2017-12-25T22:00:00.000Z,
        ],
        "reserved": Array [
          Object {
            "from": 2017-12-25T22:00:00.000Z,
            "to": 2017-12-27T22:00:00.000Z,
          },
        ],
        "selectionEnd": undefined,
        "selectionStart": undefined,
        "startingDay": Array [
          2017-12-09T22:00:00.000Z,
          2017-12-14T22:00:00.000Z,
          2017-12-19T22:00:00.000Z,
        ],
      }
    }
    navbarElement={
      <Navbar
        classNames={
          Object {
            "body": "DayPicker-Body",
            "caption": "DayPicker-Caption",
            "container": "DayPicker",
            "day": "DayPicker-Day",
            "disabled": "disabled",
            "footer": "DayPicker-Footer",
            "interactionDisabled": "DayPicker--interactionDisabled",
            "month": "DayPicker-Month",
            "months": "DayPicker-Months",
            "navBar": "DayPicker-NavBar",
            "navButtonInteractionDisabled": "DayPicker-NavButton--interactionDisabled",
            "navButtonNext": "DayPicker-NavButton DayPicker-NavButton--next",
            "navButtonPrev": "DayPicker-NavButton DayPicker-NavButton--prev",
            "outside": "outside",
            "selected": "selected",
            "today": "today",
            "todayButton": "DayPicker-TodayButton",
            "week": "DayPicker-Week",
            "weekNumber": "DayPicker-WeekNumber",
            "weekday": "DayPicker-Weekday",
            "weekdays": "DayPicker-Weekdays",
            "weekdaysRow": "DayPicker-WeekdaysRow",
            "wrapper": "DayPicker-wrapper",
          }
        }
        dir="ltr"
        labels={
          Object {
            "nextMonth": "Next Month",
            "previousMonth": "Previous Month",
          }
        }
        showNextButton={true}
        showPreviousButton={true}
      />
    }
    numberOfMonths={2}
    onDayClick={[Function]}
    onMonthChange={[Function]}
    pagedNavigation={false}
    renderDay={[Function]}
    renderWeek={[Function]}
    reverseMonths={false}
    selectedDays={
      Object {
        "from": undefined,
        "to": undefined,
      }
    }
    showOutsideDays={false}
    showWeekDays={true}
    showWeekNumbers={true}
    tabIndex={0}
    weekdayElement={<Weekday />}
  />
</div>
`;
